staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -330,14 +334,18 @@ func (le *LeaderElector) tryAcquireOrRenew(ctx context.Context) bool {
 			klog.Errorf("error retrieving resource lock %v: %v", le.config.Lock.Describe(), err)
 			return false
 		}
-		if err = le.config.Lock.Create(ctx, leaderElectionRecord); err != nil {
-			klog.Errorf("error initially creating leader election record: %v", err)
-			return false
-		}
+		// don't create a lock, if lock creation is disabled.
+		if !le.config.DisableLockCreation {
```

Comment:
Please use 
```
if le.config.DisableLockCreation {
  return false
}
```
to avoid the nesting

Reply:
Let's also add a `klog.Errorf` message. No process can become the leader if the lock doesn't exist. If a user forgets to create the lock when `DisableLockCreation` is true, it's their responsibility, but let's at least make the issue more obvious to the user.

Reply:
I think a `klog.Warningf` could be a better log level for this, since there's technically no error, it's improper use of the api.

Reply:
Agreed